Assets/MixedRealityToolkit.Services/InputSystem/GazePointerStateMachine.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{ 
+    /// <summary>
+    /// Helper class for managing the visibility of the gaze pointer to match windows mixed reality and HoloLens 2
+    /// When application starts, gaze pointer is visible. Then when articulate hands / motion controllers
+    /// appear, hide the gaze cursor. Whenever user says the voice wake word, make the 
+    /// gaze controller appear.
+    /// </summary>
+    class GazePointerStateMachine : IMixedRealitySpeechHandler
```

Comment:
Nit: The name threw me on this one because it sounds like a generic pointer utility. Maybe GazePointerSpeechStateMachine?

Reply:
Renamed to GazePointerVisibiiltyStateMachine